pilot/pkg/config/kube/gateway/gatewayclass.go

```diff
@@ -83,6 +83,9 @@ func (c *ClassController) reconcileClass(class gateway.ObjectName) error {
 	gc := &gateway.GatewayClass{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: string(class),
+			Labels: map[string]string{
+				"operator.istio.io/component": "gateway",
```

Comment:
I want istioctl to be able to recognize this gateway resource here, and then delete it.

Although I know this is very hacky. 😂

Reply:
/cc @jacob-delgado @hanxiaop PTAL.

I want to fix this issue, can you help me?

Reply:
Is there any unexpected result if it's not being deleted?

Reply:
I have no good option since it's not generated by istioctl. `istioctl uninstall `only deletes things that are installed by istioctl, and the gateway class is not one of them.

Reply:
>. Is there any unexpected result if it's not being deleted?

This may leave some end users puzzled: even without installing Istio in k8s, why can Istio still provide gateway services?

Reply:
/cc @howardjohn

Reply:
Yeah I suppose with --purge it seems reasonable to remove it...

We also write `istio-ca-root-cert` though, I don't think that is removed? Seems we should be consistent. 

I don't mind the change but we should not put operator.istio.io labels on it; its not actually operator and "operator" is kind of going away in some form so it feels quite awkward

Reply:
Thank you for your reply.

> I don't mind the change but we should not put operator.istio.io labels on it; its not actually operator and "operator" is kind of going away in some form so it feels quite awkward


So, do you have any suggestions? Should it be set like `istio-ca-root-cert` to `istio.io/config: true`?

Reply:
`istio.io/config: true` won't work because purging is similar to a normal uninstall, which also relies on operator labels. If we want to delete the gatewayclass or things like `istio-ca-root-cert`, we may need to design a logic that better detects these resources during purging instead of just adding the operator label here.